File: apps/api/app/schemas/telemetry.py
from pydantic import Field

from app.schemas.base import CamelModel


class TelemetryEventRequest(CamelModel):
    anonymous_id: str = Field(alias="anonymousId")
    event_name: str = Field(alias="eventName")
    source: str | None = None
    properties: dict[str, str | int | bool | None] | None = None


class TelemetryEventResponse(CamelModel):
    accepted: bool
    event_id: str = Field(alias="eventId")
    created_at: str = Field(alias="createdAt")


class TelemetrySummaryResponse(CamelModel):
    workspace_scans: int = Field(alias="workspaceScans")
    fixes_applied: int = Field(alias="fixesApplied")
    reviews_generated: int = Field(alias="reviewsGenerated")
    reviews_published: int = Field(alias="reviewsPublished")

File: packages/shared-types/telemetry.ts
export type TelemetryEventName =
  | "app_activated"
  | "scan_completed"
  | "workspace_scan"
  | "review_completed"
  | "review_generated"
  | "review_published"
  | "fix_applied"
  | "finding_clicked"
  | "feature_used";

export interface TelemetryProperties {
  [key: string]: string | number | boolean | null | undefined;
}

export interface TelemetryEventRequest {
  anonymousId: string;
  eventName: TelemetryEventName;
  source?: string;
  properties?: TelemetryProperties;
}

export interface TelemetryEventResponse {
  accepted: boolean;
  eventId: string;
  createdAt: string;
}

export interface TelemetrySummaryResponse {
  workspaceScans: number;
  fixesApplied: number;
  reviewsGenerated: number;
  reviewsPublished: number;
}
